coordinator/gscoordinator/kubernetes_launcher.py

```diff
@@ -353,13 +425,90 @@ def close_vineyard(self):
         # Use delete deployment instead
         pass
 
-    def create_interactive_instance(self, object_id: int, schema_path: str):
+    def check_if_engine_exist(self, engine_type):
+        """Checks if the engine with the given type exists.
+
+        Args:
+            engine_type: The type of engine to check for.
+
+        Returns:
+            True if the engine exists, False otherwise.
+        """
+
+        engine_pod_name_list = getattr(self, f"_{engine_type}_pod_name_list")
+        engine_pod_ip_list = getattr(self, f"_{engine_type}_pod_ip_list")
+        engine_pod_host_ip_list = getattr(self, f"_{engine_type}_pod_host_ip_list")
+
+        return (
+            engine_pod_name_list is not None
```

Comment:
The default value of these lists `_{engine_type}_pod_name_list` are `[]`, which is not None. So by default it's always not None. Is this intended?